deployment/probeinstaller.go

```diff
@@ -0,0 +1,67 @@
+package deployment
+
+import (
+    "context"
+    "errors"
+    "fmt"
+    "strings"
+
+    "github.com/kubearmor/kubearmor-client/k8s"
+    "github.com/kubearmor/kubearmor-client/install"
+    metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// Options for probe daemonset options install
+type ProbeOptions struct {
+    Namespace      string
+    ProbeDaemonImage string
+    Audit          string
+    Force          bool
+}
+
+// K8sInstaller for karmor install
+func ProbeDaemonInstaller(c *k8s.Client, o ProbeOptions) error {
+    env := install.AutoDetectEnvironment(c)
+    if env == "none" {
+        return errors.New("unsupported environment or cluster not configured correctly")
+    }
+    // fmt.Printf("Auto Detected Environment : %s\n", env)
+
+    daemonset := GenerateDaemonSet(env, o.Namespace)
+    daemonset.Spec.Template.Spec.Containers[0].Image = o.ProbeDaemonImage
+    if o.Audit == "all" || strings.Contains(o.Audit, "file") {
+        daemonset.Spec.Template.Spec.Containers[0].Args = append(daemonset.Spec.Template.Spec.Containers[0].Args, "-defaultFilePosture=audit")
+    }
+    if o.Audit == "all" || strings.Contains(o.Audit, "network") {
+        daemonset.Spec.Template.Spec.Containers[0].Args = append(daemonset.Spec.Template.Spec.Containers[0].Args, "-defaultNetworkPosture=audit")
+    }
+    if o.Audit == "all" || strings.Contains(o.Audit, "capabilities") {
+        daemonset.Spec.Template.Spec.Containers[0].Args = append(daemonset.Spec.Template.Spec.Containers[0].Args, "-defaultCapabilitiesPosture=audit")
+    }
+    // fmt.Printf("Probe DaemonSet %s %v...\n", daemonset.Spec.Template.Spec.Containers[0].Image, daemonset.Spec.Template.Spec.Containers[0].Args)
+
```

Comment:
Why do we need these here 😅 ?

I believe these are specific to KubeArmor. We shouldn't be needing to detect an environment since Probe will be the one detecting environment for us or modifying the Image and Args...